web-services-app/src/main/java/edu/unc/lib/boxc/web/services/processing/IiifV3ManifestService.java

```diff
@@ -215,27 +224,42 @@ private void setImageContent(ContentObjectRecord contentObj, PaintingAnnotation
 
     private void assignImageDimensions(ContentObjectRecord contentObj, Canvas canvas, ImageContent imageContent) {
         var dimensions = getDimensions(contentObj);
-        var width = Integer.parseInt(dimensions.get(WIDTH));
-        var height = Integer.parseInt(dimensions.get(HEIGHT));
-        canvas.setWidthHeight(width, height); // Dimensions for the canvas
-        imageContent.setWidthHeight(width, height); // Dimensions for the actual image
+        if (dimensions != null) {
+            var width = dimensions.get(WIDTH);
+            var height = dimensions.get(HEIGHT);
+            canvas.setWidthHeight(width, height); // Dimensions for the canvas
+            imageContent.setWidthHeight(width, height); // Dimensions for the actual image
+        }
     }
 
-    private HashMap<String, String> getDimensions(ContentObjectRecord contentObj) {
-        var dimensions = new HashMap<String,String>();
+    private Map<String, Integer> getDimensions(ContentObjectRecord contentObj) {
         var fileDs = getFileDatastream(contentObj);
         String extent = fileDs.getExtent();
         if (extent != null && !extent.isEmpty()) {
             String[] imgDimensions = extent.split("x");
-            // height x width x seconds
-            var height = imgDimensions[0];
-            var width = imgDimensions[1];
-            dimensions.put(WIDTH, width);
-            dimensions.put(HEIGHT, height);
-            if (Arrays.stream(imgDimensions).count() == 3) {
-                var duration = imgDimensions[2];
-                dimensions.put(DURATION, duration);
+            var imgDimensionsCount = Arrays.stream(imgDimensions).count();
+            if (imgDimensionsCount < 2) {
+                return null;
             }
+            return extractDimensions(imgDimensions, imgDimensionsCount);
+        }
+        return null;
```

Comment:
Two suggestions here:
1) We should change it to using the length value rather than streaming and counting.
2) If we flip the conditional to >= 2, then we don't need the extra `return null` and can just use the one at the end of the method.
```
            if (imgDimensions.length >= 2) {
                return extractDimensions(imgDimensions, imgDimensionsCount);
            }
        }
        return null;
```